src/core/internal/dassert.d

```diff
@@ -95,10 +95,32 @@ private string miniFormat(V)(const scope ref V v)
     }
     else static if (__traits(isIntegral, V))
     {
-        enum printfFormat = getPrintfFormat!V;
-        char[20] val;
-        const len = sprintf(&val[0], printfFormat, v);
-        return val.idup[0 .. len];
+        static if (is(V == char))
+        {
+            // Avoid invalid code points
+            if (v < 0x7F)
+                return ['\'', v, '\''];
+
+            return "cast(char) " ~ miniFormat(*cast(ubyte*) &v);
```

Comment:
I'm not in love with this reinterpret cast. It wouldn't be horrible to create a tmp as in the other branch.